### PR TITLE
Fix occ app management for group apps

### LINKED ISSUE #17737: Apps that are enabled for certain groups only are shown as disabled at `php occ app:list`
### Steps to reproduce
1. Enable app for only one group, for example "admin". (In table `oc_appconfig` the configValue for `enabled` would be `["admin"]`, everything is ok if it is `yes`.)
2. Call `php occ app:list`.

### Expected behaviour
App should be shown as enabled.

### Actual behaviour
App is shown as disabled.

### Server configuration
**Operating system**:
Ubuntu Linux 15.04
**Web server:**
Apache
**Database:**
MySQL
**PHP version:**
5.6.4
**ownCloud version:** (see ownCloud admin page)
8.2 pre alpha
**Updated from an older ownCloud or fresh install:**
update
**List of activated apps:**
Enabled:
  - bookmarks: 0.4
  - calendar: 0.7.2
  - contacts: 0.4.0.0
  - files: 1.1.9
  - files_external: 0.2.3
  - files_sharing: 0.6.2
  - files_trashbin: 0.6.3
  - files_versions: 1.0.6
  - news: 5.3.8
  - tasks: 0.7.1

Disabled:
  - encryption
  - ownbackup
  - provisioning_api
  - sharelinks
  - user_ldap
  - user_webdavauth

**The content of config/config.php:**
```json
{
    "system": {
        "instanceid": "ociuf2r3opzp",
        "trusted_domains": [
            "localhost"
        ],
        "datadirectory": "\/home\/omega\/Web\/owncloud\/data",
        "overwrite.cli.url": "http:\/\/localhost:8080",
        "dbtype": "mysql",
        "version": "8.2.0.2",
        "dbname": "owncloud",
        "dbhost": "localhost",
        "dbtableprefix": "oc_",
        "logtimezone": "UTC",
        "installed": true,
        "theme": "",
        "maintenance": false
    }
}
```

**Are you using external storage, if yes which one:** local/smb/sftp/...
no

**Are you using encryption:** yes/no
no

**Are you using an external user-backend, if yes which one:** LDAP/ActiveDirectory/Webdav/...
no

#### Web server error log
nothing was logged

#### ownCloud log (data/owncloud.log)
nothing was logged

### PR DESCRIPTION
**Pull request by nickvergessen**
### 1. App enabled for a group only, is not listed as "enabled"
1. Enable an app for a group only
2. Run `./occ app:list`
### 2. Can not disable app that is enabled for a specific group
1. Enable an app for a group only
2. Run `./occ app:disable testing`
### 3. Can not enable app for a specific group
1. Run `./occ app:enable help`

cc @MorrisJobke @LukasReschke @DeepDiver1975 

Fix #17737

### CODE REVIEW

**Comment by nickvergessen** [comment]
I'd vote for something like `\OC_App::isEnabled($appId, false)` to be supported. (User false)
Problem is User null uses the current user, in occ thats no one, so there are no groups, so no "group enabled" apps are enabled. Now we are using the config to check it, instead of having that way supported by our API.

File: core/command/app/listapps.php [hunk 1]
```diff
@@ -59,10 +71,10 @@ protected function execute(InputInterface $input, OutputInterface $output) {
 
 		//sort enabled apps above disabled apps
 		foreach ($apps as $app) {
-			if ($shouldFilterShipped && \OC_App::isShipped($app) !== $shippedFilter){
+			if ($shippedFilter !== null && \OC_App::isShipped($app) !== $shippedFilter){
 				continue;
 			}
-			if (\OC_App::isEnabled($app)) {
+			if ($this->config->getAppValue($app, 'enabled', 'no') !== 'no') {
```
Comment (PVince81) [inline core/command/app/listapps.php]: @nickvergessen you mean use "isEnabled($app, false)" here if it was supported ?

Comment (nickvergessen) [inline core/command/app/listapps.php]: yes,

**Comment by icewind1991** [comment]
```
\OC::$server->getAppManager()->isInstalled($appId);
```

**Comment by nickvergessen** [comment]
> ```php
\OC::$server->getAppManager()->isInstalled($appId);
```

Ah okay, so it's just a confusing method name (because install is not enable)

Fixed to use that

**Comment by MorrisJobke** [comment]
@nickvergessen I guess this fixes also #17737 ;)

**Comment by MorrisJobke** [comment]
Tested and works :+1:

**Comment by pbek** [comment]
nice :+1:

**Comment by nickvergessen** [comment]
Yes it does @MorrisJobke

**Comment by DeepDiver1975** [comment]
rebased - I want to the issues on external storage

**Comment by MorrisJobke** [comment]
Really weird stuff:


```
OCA\Files_External\Tests\Service\DBConfigServiceTest::testAddSimpleMount
Doctrine\DBAL\Exception\TableNotFoundException: An exception occurred while executing 'INSERT INTO "oc_external_mounts" ("mount_point", "storage_backend", "auth_backend", "priority", "type") VALUES(?, ?, ?, ?, ?)' with params ["\/test", "foo", "bar", 100, 1]:

SQLSTATE[HY000]: General error: 1 no such table: oc_external_mounts
```

**Comment by DeepDiver1975** [comment]
I guess this is because the files_external app is not properly enabled

**Comment by PVince81** [comment]
Possibly similar to https://github.com/owncloud/core/issues/21784
Some parts of the FS code seem to assume that files_external is enabled.
CC @icewind1991

**Comment by DeepDiver1975** [comment]
I assume the issue is within this PR - havn't seen this on other prs - in addition only external unit tests fail which requires files_external to be enabled

**Comment by nickvergessen** [comment]
Okay, will have a look later

File: core/command/app/enable.php [hunk 2]
```diff
@@ -37,19 +51,36 @@ protected function configure() {
...
-			\OC_App::enable($appId);
+		}
+
+		$groups = $input->getOption('groups');
+		if ($this->manager->isInstalled($appId) && empty($groups)) {
+			$output->writeln($appId . ' is already enabled');
+		}
+
+		if (empty($groups)) {
+			$this->manager->enableApp($appId);
```
Comment (nickvergessen) [inline core/command/app/enable.php]: This is what broke the ext. sto. tests...

`\OC_App::enable()` also runs install, while the appmanager seems to not do that. So your instance will be in two different states, depending on whether the first enable is via the appmanager or oc_app.

Will revert to OC_App for now

**Comment by DeepDiver1975** [comment]
:+1:

**Pull request merged**